Move internal CI manifest to nonfree

sermazz opened the pull request:
This PR moves the internal GitLab CI manifest to the nonfree component, for easier future usage with proprietary tools.
It works by launching a pipeline in the internal repo https://iis-git.ee.ethz.ch/github-mirror/chimera. Within this, the repo https://iis-git.ee.ethz.ch/pulp-restricted/chimera-nonfree is cloned and the CI subpipe contained there is spawned. This streamlines the usage of scripts for licensed tools and technology files, which will all be contained in `chimera-nonfree`.
This PR also addresses the comments in https://github.com/pulp-platform/chimera/pull/14.

micprog commented on chimera.mk:
For the future, one option to avoid excessive updates of this commit hash when updating the internal CI is to use a tag that gets moved in the internal repository (see iDMA), but I think this is fine for now.

sermazz replied:
Also did this, since I was at it.

micprog reviewed (approved):
Looks like it works, so LGTM, thanks for incorporating the fixes! :+1:

sermazz commented on .gitlab-ci.yml:
```diff
@@ -1,86 +1,25 @@
-  needs: [ vsim-build ]
-  parallel:
-    matrix:
-      - TESTNAME: [testCluster, testClusterOffload]
```
Done! I think it's better to still decide manually which tests to execute. I also made it in a generic way in case the same approach is required for different usage in the future.
Now there's a `.gitlab` directory which contains the internal GitLab CI manifest (launching the nonfree subpipe), and another yml including only a list of variables which are used by the CI in the nonfree repo.

sermazz commented:
Comments addressed. Just for completeness, I list here the required changes in the other repositories:
- https://iis-git.ee.ethz.ch/pulp-restricted/chimera-nonfree: I created a `deploy` tag which is cloned instead of the commit hash; also added a readme to explain its usage
- https://iis-git.ee.ethz.ch/github-mirror/chimera: I changed the path of the CI to execute to `.gitlab/gitlab-ci.yml`, as I created this new directory to also hold `nonfree-var.yml` and keep stuff tidy. The internal CI might not work anymore for branches which do not include the commits in this PR

Scheremo commented:
> I addressed your review :) Just for completeness, I list here the required changes in the other repositories:
> 
>     * https://iis-git.ee.ethz.ch/pulp-restricted/chimera-nonfree: I created a `deploy` tag which is cloned instead of the commit hash; also added a readme to explain its usage
> 
>     * https://iis-git.ee.ethz.ch/github-mirror/chimera: I changed the path of the CI to execute to `.gitlab/gitlab-ci.yml`, as I created this new directory to also hold `nonfree-var.yml` and keep stuff tidy. The internal CI might not work anymore for branches which do not include the commits in this PR

Looks good to me! Thanks for the support for test additions in the main repo.
